frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMachetePanel.java

```diff
@@ -42,6 +47,19 @@ public GitMachetePanel(Project project) {
     val selectionComponent = selectedGitRepositoryProvider.getSelectionComponent();
     val graphTable = getGraphTable();
 
+    project.getMessageBus().connect().subscribe(VirtualFileManager.VFS_CHANGES, new BulkFileListener() {
+      @Override
+      public void after(java.util.List<? extends VFileEvent> events) {
+        for (val event : events) {
+          if (event instanceof VFileContentChangeEvent) {
```

Comment:
Fun fact, Java 16 would allow for:

`if (event instanceof VFileContentChangeEvent changeEvent) {`

where `changeEvent` is of type `VFileContentChangeEvent`, not `VFileEvent` (so a downcast below won't be necessary)